### PR TITLE
Add Create Account fromDerivePath

### PR DESCRIPTION
**Pull request by daoauth**
### Description
Create Account with [hdpath 637](https://github.com/satoshilabs/slips/pull/1364) 

### Test Plan
```
test('generates derive path accounts', () => {
  const address = '0x07968dab936c1bad187c60ce4082f307d030d780e91e694ae03aef16aba73f30';
  const a1 = AptosAccount.fromDerivePath('m/44\'/637\'/0\'/0\'/0\'', mnemonic);
  expect(a1.address().hex()).toBe(address);
});

test('generates derive path accounts', () => {
  expect(() => { AptosAccount.fromDerivePath('', mnemonic); }).toThrow(new Error('Invalid derivation path'));
});
```

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2073)

### CODE REVIEW

**Comment by davidiw** [comment]
Great stuff! Lemme get some typescript ninjas on this.

File: ecosystem/typescript/sdk/package.json [hunk 1]
```diff
@@ -51,7 +51,9 @@
   },
   "dependencies": {
     "axios": "^0.26.1",
+    "bip39-light": "^1.0.7",
```
Comment (hariria) [inline ecosystem/typescript/sdk/package.json:54]: https://github.com/paulmillr/scure-bip39 might be a better fit. Runs in the browser without issue

Comment (hariria) [reply ecosystem/typescript/sdk/package.json:54]: Used in [js-ethereum-cryptography](https://github.com/ethereum/js-ethereum-cryptography/blob/master/package.json)

File: ecosystem/typescript/sdk/src/aptos_account.ts [hunk 2]
```diff
@@ -30,6 +32,39 @@ export class AptosAccount {
     return new AptosAccount(HexString.ensure(obj.privateKeyHex).toUint8Array(), obj.address);
   }
 
+  static isValidPath = (path: string): boolean => {
+    if (
+      // eslint-disable-next-line prefer-regex-literals
+      !new RegExp('^m\\/44+\'\\/637+\'\\/[0-9]+\'\\/[0-9]+\'\\/[0-9]+\'+$').test(
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/aptos_account.ts:38]: This doesn't look right:
1. First of all, rebase and start using double quotes
2. Let's follow the BIP44 standard, https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#path-levels. The `change` and `address_index` do not follow by a `'`. In the link, a path looks like `m / purpose' / coin_type' / account' / change / address_index`.
3. The regex has issues. `44+` would match `44444` and `637+` would match `63777777`. 44 and 637 should be an exact match. 44 is the current BIP number and 637 is Apto's coin type.

File: ecosystem/typescript/sdk/src/aptos_account.ts [hunk 3]
```diff
@@ -30,6 +32,39 @@ export class AptosAccount {
...
+      )
+    ) {
+      return false;
+    }
+    return !path
+      .split('/')
+      .slice(1)
+      .map((val: string): string => val.replace('\'', ''))
+      // eslint-disable-next-line no-restricted-globals
+      .some(isNaN as any);
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/aptos_account.ts:49]: Why is this needed, if the regex worked?

File: ecosystem/typescript/sdk/src/aptos_account.ts [hunk 4]
```diff
@@ -30,6 +32,39 @@ export class AptosAccount {
...
+  };
+
+  static fromDerivePath(path: string, mnemonics: string): AptosAccount {
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/aptos_account.ts:52]: Also, check if mnemonics is empty and throw meaning full errors, just in case developers forgot to pass in the mnemonics.

File: ecosystem/typescript/sdk/src/types/bip39-light.d.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+declare module 'bip39-light' {
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/types/bip39-light.d.ts:1]: Why do you have to add the types manually? It could not be installed as a dependency?

Comment (daoauth) [reply ecosystem/typescript/sdk/src/types/bip39-light.d.ts:1]: `@types/bip39-light` is not exist. but I'll replace to `scure-bip39` and remove this.

File: ecosystem/typescript/sdk/src/aptos_account.ts (hunk 3 above)
Comment (daoauth) [reply ecosystem/typescript/sdk/src/aptos_account.ts:49]: yes. I'll remove it.

File: ecosystem/typescript/sdk/src/aptos_account.ts (hunk 2 above)
Comment (daoauth) [reply ecosystem/typescript/sdk/src/aptos_account.ts:38]: Because it uses the [ed25519-hd-key module](https://github.com/alepop/ed25519-hd-key/blob/master/src/utils.ts), the regex check uses a `'` for all numbers. And other chains, [such as near](https://github.com/near/near-ledger-js/blob/67d5859a775290ae9bacec6bf2e2ccf6009dce30/demo/index.html#L13-L18) and [solana](https://medium.com/@josh.wolff.7/derivation-paths-on-solana-fb08d3dd09f1) are using a `'` for all numbers.

File: ecosystem/typescript/sdk/src/aptos_account.ts [hunk 6]
```diff
@@ -30,6 +32,34 @@ export class AptosAccount {
     return new AptosAccount(HexString.ensure(obj.privateKeyHex).toUint8Array(), obj.address);
   }
 
+  static isValidPath = (path: string): boolean => {
+    if (
+      // eslint-disable-next-line prefer-regex-literals
+      !new RegExp("^m\\/44'\\/637'\\/[0-9]+'\\/[0-9]+'\\/[0-9]+'+$").test(
+        path,
+      )
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/aptos_account.ts:40]: Why not one line?

File: ecosystem/typescript/sdk/src/aptos_account.ts [hunk 7]
```diff
@@ -30,6 +32,34 @@ export class AptosAccount {
     return new AptosAccount(HexString.ensure(obj.privateKeyHex).toUint8Array(), obj.address);
   }
 
+  static isValidPath = (path: string): boolean => {
+    if (
+      // eslint-disable-next-line prefer-regex-literals
+      !new RegExp("^m\\/44'\\/637'\\/[0-9]+'\\/[0-9]+'\\/[0-9]+'+$").test(
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/aptos_account.ts:38]: You can either:
`new RegExp("^m/44'/637'/[0-9]+'/[0-9]+'/[0-9]+'+$").test` Or `/^m\/44'\/637'\/[0-9]+'\/[0-9]+'\/[0-9]+'+$/.test`

Comment (jjleng) [reply ecosystem/typescript/sdk/src/aptos_account.ts:38]: The `\\` is not needed

File: ecosystem/typescript/sdk/src/aptos_account.ts (hunk 4 above)
Comment (jjleng) [reply ecosystem/typescript/sdk/src/aptos_account.ts:47]: You still need to check if the mnemonics are empty. Why should we allow that to happen? Throw an error in such case.

File: ecosystem/typescript/sdk/src/aptos_account.ts [hunk 8]
```diff
@@ -30,6 +32,34 @@ export class AptosAccount {
     return new AptosAccount(HexString.ensure(obj.privateKeyHex).toUint8Array(), obj.address);
   }
 
+  static isValidPath = (path: string): boolean => {
```
Comment (jjleng) [inline ecosystem/typescript/sdk/src/aptos_account.ts:35]: Add @ts-doc for this method

File: ecosystem/typescript/sdk/src/aptos_account.ts (hunk 4 above)
Comment (jjleng) [reply ecosystem/typescript/sdk/src/aptos_account.ts:47]: Add ts-doc for this method

**Review by jjleng — COMMENTED** [review]
Please squash your commits and make sure your final commit message follows the Angular Conventional Commits. This is to help the SDK tooling find your commit and add it to release notes.

For your final commit, the message should look like this:
feat: add BIP44 to allow the creation of multiple accounts with the same key

File: ecosystem/typescript/sdk/src/aptos_account.ts (hunk 6 above)
Comment (daoauth) [reply ecosystem/typescript/sdk/src/aptos_account.ts:40]: I'll fix it

File: ecosystem/typescript/sdk/src/aptos_account.ts (hunk 7 above)
Comment (daoauth) [reply ecosystem/typescript/sdk/src/aptos_account.ts:38]: I'll fix it

**Comment by daoauth** [comment]
@jjeng [@scure/bip39](https://github.com/paulmillr/scure-bip39/blob/014f4a5d5d982adbbcc1119451175b66b93f3681/src/index.ts#L20-L25)
```javascript
function normalize(str: string) {
  const norm = nfkd(str);
  const words = norm.split(' ');
  if (![12, 15, 18, 21, 24].includes(words.length)) throw new Error('Invalid mnemonic');
  return { nfkd: norm, words };
}
```
Length check of mnemonics is in the @scure/bip39 module.

**Review by jjleng — APPROVED** [review]
Thanks

**Comment by gregnazario** [comment]
I think you need to rebase, there's a ton of commits here that are already in the codebase

**Comment by davidiw** [comment]
please update, it seems like this is stale

**Comment by daoauth** [comment]
@davidiw update was done thanks.

File: ecosystem/typescript/sdk/package.json [hunk 9]
```diff
@@ -39,6 +39,8 @@
     "tweetnacl": "^1.0.3",
     "typescript-memoize": "^1.1.0",
     "yarn": "^1.22.19"
```
Comment (jjleng) [inline ecosystem/typescript/sdk/package.json:41]: You missed a comma here.

Comment (alinush) [reply ecosystem/typescript/sdk/package.json]: I fixed it.

Comment (alinush) [reply ecosystem/typescript/sdk/package.json]: Weird, that broke something:

```
Run cd ./ecosystem/typescript/sdk/examples/typescript && yarn install
yarn install v1.22.1[9](https://github.com/aptos-labs/aptos-core/runs/7983140233?check_suite_focus=true#step:4:10)
error An unexpected error occurred: "/home/runner/work/aptos-core/aptos-core/ecosystem/typescript/sdk/package.json: Unexpected token } in JSON at position 1252".
info If you think this is a bug, please open a bug report with the information provided in "/home/runner/work/aptos-core/aptos-core/ecosystem/typescript/sdk/examples/typescript/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Error: Process completed with exit code 1.
```
